### PR TITLE
Add support for accept attribute on file input.

### PR DESCRIPTION
**Pull request by thet**
If the widget's field - if there is one - has  set (the  field has  set by default) the allowed content types are rendered as  attribute on the file input.

This already restricts the allowed file types before uploading while still being checked on the server side too.

Fixes: https://github.com/plone/plone.formwidget.namedfile/issues/66 Depeds on:
- https://github.com/plone/plone.namedfile/pull/158
- https://github.com/plone/plone.formwidget.namedfile/pull/67

### CODE REVIEW

**Comment by thet** [comment]
@jenkins-plone-org please run jobs

File: plone/app/z3cform/templates/file_input.pt [hunk 1]
```diff
@@ -90,13 +90,21 @@
 
   <input class="form-control ${python:view.error and 'is-invalid' or ''}"
          id="${view/id}-input"
+         accept="${view/accept}"
```
Comment (mauritsvanrees) [inline plone/app/z3cform/templates/file_input.pt:93]:
```suggestion
         accept="${view/accept|nothing}"
```

File: plone/app/z3cform/templates/file_input.pt [hunk 2]
```diff
@@ -90,13 +90,21 @@
...
          disabled="${view/disabled}"
          maxlength="${view/maxlength}"
          name="${view/name}"
          required="${python:view.required and 'required' or None}"
          size="${view/size}"
          type="file"
   />
+  <div class="form-text"
+       tal:condition="view/accept"
```
Comment (mauritsvanrees) [inline plone/app/z3cform/templates/file_input.pt:102]:
```suggestion
       tal:condition="view/accept|nothing"
```

File: plone/app/z3cform/templates/image_input.pt [hunk 3]
```diff
@@ -94,13 +94,21 @@
 
   <input class="form-control ${python:view.error and 'is-invalid' or ''}"
          id="${view/id}-input"
+         accept="${view/accept}"
```
Comment (mauritsvanrees) [inline plone/app/z3cform/templates/image_input.pt:97]:
```suggestion
         accept="${view/accept|nothing}"
```

File: plone/app/z3cform/templates/image_input.pt [hunk 4]
```diff
@@ -94,13 +94,21 @@
...
          disabled="${view/disabled}"
          maxlength="${view/maxlength}"
          name="${view/name}"
          required="${python:view.required and 'required' or None}"
          size="${view/size}"
          type="file"
   />
+  <div class="form-text"
+       tal:condition="view/accept"
```
Comment (mauritsvanrees) [inline plone/app/z3cform/templates/image_input.pt:106]:
```suggestion
       tal:condition="view/accept|nothing"
```

**Review by mauritsvanrees — CHANGES_REQUESTED** [review]
I tried it out and it works in combination with the other PRs.

I have added suggestions to add `|nothing` so it does not break when this code is used in combination with a field/widget that does not have an `accept` attribute.  Maybe someone is using this on a field that we do not expect.

With those `|nothing` added, the input still works when I try it with an old release of `plone.formwidget.namedfile`.

**Review by mauritsvanrees — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
I have released a new `plone.formwidgets.namedfile` and depend on it.  It is up in the constraints on dist.plone.org.  Tests pass now.

But the dependencies check fails.  Should have nothing to do with this PR.  I am on it, trying an update with `plone/meta`.

**Comment by mauritsvanrees** [comment]
All GitHub Actions pass.  Now try Jenkins again:

@jenkins-plone-org please run jobs

**Pull request merged**